packages/core/ios/Sources/DdSdk.swift

```diff
@@ -22,7 +22,16 @@ class RNDdSdk: NSObject {
 
     @objc(initialize:withResolver:withRejecter:)
     func initialize(configuration: NSDictionary, resolve:RCTPromiseResolveBlock, reject:RCTPromiseRejectBlock) -> Void {
-        nativeInstance.initialize(configuration: configuration.asDdSdkConfiguration())
+        // Datadog SDK init needs to happen on the main thread: https://github.com/DataDog/dd-sdk-reactnative/issues/198
+        // This is safe because the main queue will never be blocked waiting for the sharedQueue
+        // Adding an extra check in case sharedQueue is ever changed to be DispatchQueue.main.
+        if Thread.isMainThread {
+            nativeInstance.initialize(configuration: configuration.asDdSdkConfiguration())
+        } else {
+            DispatchQueue.main.sync {
+                nativeInstance.initialize(configuration: configuration.asDdSdkConfiguration())
+            }
+        }
         resolve(nil)
```

Comment:
The extra check is required to avoid deadlock, I'm not familiar with RN threading but since the method signature has callbacks it seems safer to do an `async` dispatch here:
```suggestion
       DispatchQueue.main.async {
          nativeInstance.initialize(configuration: configuration.asDdSdkConfiguration())
          resolve(nil)
      }
```

Would that work?